### PR TITLE
Added protocol in response to Issue #145

### PR DESCRIPTION
**Pull request by MrJester**
**Before submitting a pull request, make sure to complete the following:**
- [X] Ensure there are no similar pull requests.
- [X] Read the [Development Guide](https://github.com/lanmaster53/recon-ng/wiki/Development-Guide).

**What kind of PR is this?**  
_Please add an 'x' in the appropriate box, and apply a label to the PR matching the type here._
- [X] Bug Fix
- [ ] New Module
- [ ] Documentation Update

**Checklist For Approval**
- [X] Updated the meta dictionary for the module.
  - If bug fix, updated the version.
- [ ] Indexed the module
- [ ] Added the index to the `modules.yml` file
- [X] Made the most out of the available [mixins](https://github.com/lanmaster53/recon-ng/wiki/Development-Guide#mixins).
- [X] Ensured the code is PEP8 compliant with `pycodestyle` or `black`.

### CODE REVIEW

**Comment by MrJester** [comment]
So I started just adding the protocol to the DB Insert then I decided to address two issues in one and flip this over to the using the shodan python module. Does appear to work a lot faster and I have not run into any of the timeout issues running it on some larger data sets.

**Comment by lanmaster53** [comment]
Cool. Why the version hop from 1.0 to 1.2? What happened to 1.1? Just a typo?

File: modules.yml [hunk 1]
```diff
@@ -851,16 +851,17 @@
...
     operator. Updates the 'ports' table with the results.
   files: []
-  last_updated: '2019-06-24'
+  last_updated: '2020-06-21'
   name: Shodan IP Enumerator
   path: recon/hosts-ports/shodan_ip
   required_keys:
   - shodan_api
-  version: '1.0'
+  version: '1.2'
```
Comment (lanmaster53) [inline modules.yml:864]: Why the version hop from 1.0 to 1.2? What happened to 1.1?

File: modules/recon/hosts-ports/shodan_ip.py [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
 
-class Module(BaseModule, ShodanAPIMixin):
-
+class Module(BaseModule):
     meta = {
         'name': 'Shodan IP Enumerator',
         'author': 'Tim Tomes (@lanmaster53) and Matt Puckett (@t3lc0)',
-        'version': '1.0',
-        'description': 'Harvests port information from the Shodan API by using the \'ip\' search operator. Updates the \'ports\' table with the results.',
+        'version': '1.2',
```
Comment (lanmaster53) [inline modules/recon/hosts-ports/shodan_ip.py:10]: Why the version hop from 1.0 to 1.2? What happened to 1.1?

File: modules.yml (hunk 1 above)
Comment (MrJester) [reply modules.yml:864]: I did it in two different steps and it was muscle memory taking over.

**Comment by MrJester** [comment]
> How do you feel about upgrading all of the shodan modules?

That’s the plan

File: modules/recon/hosts-ports/shodan_ip.py [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
+            try:
+                ipinfo = api.host(ipaddr)
+                for port in ipinfo['data']:
+                    try:
+                        for hostname in port['hostnames']:
+                            self.insert_ports(host=hostname, ip_address=ipaddr, port=port['port'],
+                                              protocol=port['transport'])
+                    except KeyError:
+                        self.insert_ports(ip_address=ipaddr, port=port['port'], protocol=port['transport'])
+            except shodan.exception.APIError:
```
Comment (lanmaster53) [inline modules/recon/hosts-ports/shodan_ip.py:35]: Should we do some level of reporting here? What causes this type of error?

Comment (MrJester) [reply modules/recon/hosts-ports/shodan_ip.py:35]: The shodan module raises this error if an IP is not found in its database so we either catch it like this which how the rest of the modules seem to handle no data or we print a message saying not found in DB.

Comment (lanmaster53) [reply modules/recon/hosts-ports/shodan_ip.py:35]: OK. Cool. I like this approach then. Thanks!

**Pull request merged**